src/editor/editor.ml

```diff
@@ -0,0 +1,659 @@
+      ~container:(find_component "learnocaml-exo-toplevel-pane")
+      ~history () in
+  init_tabs () ;
+  toplevel_launch >>= fun top ->
+
+  (* ---- toplevel pane ------------------------------------------------- *)
+
+  begin toplevel_button
+      ~group: toplevel_buttons_group
+      ~icon: "cleanup" [%i"Clear"] @@ fun () ->
+    Learnocaml_toplevel.clear top ;
+    Lwt.return ()
+  end ;
+  begin toplevel_button
+      ~icon: "reload" [%i"Reset"] @@ fun () ->
+    (* toplevel_launch >>= fun top -> SHOULD BE UNNECESSARY *)
+     disabling_button_group toplevel_buttons_group
+         (fun () -> Learnocaml_toplevel.reset top)
+  end ;
+  begin toplevel_button
+      ~group: toplevel_buttons_group
+      ~icon: "run" [%i"Eval phrase"] @@ fun () ->
+    Learnocaml_toplevel.execute top ;
+    Lwt.return ()
+  end ;
+
+  (* ---- prelude pane --------------------------------------------------- *)
+
+  let editor_prelude = find_component "learnocaml-exo-prelude-pane" in
+  let editor_prel = Ocaml_mode.create_ocaml_editor
+                      (Tyxml_js.To_dom.of_div editor_prelude) in
+  let ace_prel = Ocaml_mode.get_editor editor_prel in
+  let contents= get_prelude id in
+  Ace.set_contents ace_prel contents ;
+  Ace.set_font_size ace_prel 18;
+
+  let typecheck_prelude () =
+    Editor_lib.typecheck true ace_prel editor_prel top "" (Ace.get_contents ace_prel) in
+  begin prelude_button
+      ~group: toplevel_buttons_group
+      ~icon: "typecheck" [%i"Check"] @@ typecheck_prelude
+  end;
+
+  (* ---- prepare pane --------------------------------------------------- *)
+
+  let editor_prepare = find_component "learnocaml-exo-prepare-pane" in
+  let editor_prep = Ocaml_mode.create_ocaml_editor
+                      (Tyxml_js.To_dom.of_div editor_prepare) in
+  let ace_prep = Ocaml_mode.get_editor editor_prep in
+  let contents= get_prepare id in
+  Ace.set_contents ace_prep contents ;
+  Ace.set_font_size ace_prep 18;
+
+  let typecheck_prepare () =
+    let prel = Ace.get_contents ace_prel ^ "\n" in
+    Editor_lib.typecheck true ace_prep editor_prep top prel
+      ~onpasterr:(fun () -> select_tab "prelude"; typecheck_prelude ())
+      (Ace.get_contents ace_prep) in
+  begin prepare_button
+      ~group: toplevel_buttons_group
+      ~icon: "typecheck" [%i"Check"] @@ typecheck_prepare
+  end;
+
+  (*-------question pane  -------------------------------------------------*)
+  let open Omd_representation in
+  let override_url = function
+    | Url(href,s,title) ->
+       Some ( let title_url = if title <> "" then
+                                String.concat " " [" title='";title;"'"]
+                              else "" in
+              let url =
+                String.concat
+                  ""
+                  ["<a href='";
+                   href;"' target='_blank' rel='noopener noreferrer'";
+                   title_url;">";Omd_backend.html_of_md s;"</a>"]
+              in url)
```

Comment:
Style remarks:
* is the let-in really necessary here? (you wrote `let url = … in url` :)
* Alors maybe a [Printf](https://ocaml.org/api/Printf.html) format string would be better suited for this long string concatenation?:
```suggestion
              Printf.sprintf {|<a href="%s" target="_blank" rel="noopener noreferrer"%s>%s</a>"|}
              href title_url (Omd_backend.html_of_md s)
```
A bit of context here:
- HTML indeed allows both single-quote and double-quotes for attributes:
  https://www.w3.org/TR/html4/intro/sgmltut.html#h-3.2.2
- But double-quotes are preferred:
  https://google.github.io/styleguide/htmlcssguide.html#HTML_Quotation_Marks
- In this case the OCaml syntax `{|string|}` can be handy to avoid writing `\"` within the string:
  https://stackoverflow.com/questions/63223119/nested-quotes-in-string-with-ocaml/63223231#63223231